Modules/Loadable/Markups/MRML/vtkMRMLMarkupsPlaneNode.cxx

```diff
@@ -951,6 +951,58 @@ void vtkMRMLMarkupsPlaneNode::SetSize(double x, double y)
   this->Modified();
 }
 
+//----------------------------------------------------------------------------
+void vtkMRMLMarkupsPlaneNode::GetSizeWorld(double size_World[2])
```

Comment:
vtkSlicerPlaneRepresentation3D::BuildPlane()/vtkSlicerPlaneRepresentation2D::BuildPlane() implementation computes the corner points quite differently compared from what this method does. This has a slight risk of inconsistency between what is displayed and what is computed (even if it works perfectly now, developers may make changes in the future and they may not consistenly do it in all 3 methods). Things may become non-trivial when multiple transforms are applied on the plane (including warping) and/or the bounds of the plane are shifted (so that the plane center is not the same as the control point position), etc.

Could you refactor the computations so that all 3 methods to ensure their consistency? For example, a GetPlaneCornerPointsWorld() method could be added to vtkMRMLMarkupsPlaneNode that would compute the corner positions and that would be used in GetSizeWorld and BuildPlane methods. This new method might also simplify other methods, such as GetRASBounds, GetBounds, CalculatePlaneBounds (and ensure consistency).